src/wayland/foreign_toplevel_list/mod.rs

```diff
@@ -0,0 +1,499 @@
+//! Foreign toplevel list
+//!
+//! The purpose of this protocol is to provide protocol object handles for toplevels.
+//!
+//! ```no_run
+//! use smithay::wayland::foreign_toplevel_list::{ForeignToplevelListState, ForeignToplevelListHandler};
+//!
+//! pub struct State {
+//!     foreign_toplevel_list: ForeignToplevelListState,
+//! }
+//!
+//! smithay::delegate_foreign_toplevel_list!(State);
+//!
+//! impl ForeignToplevelListHandler for State {
+//!     fn foreign_toplevel_list_state(&mut self) -> &mut ForeignToplevelListState {
+//!         &mut self.foreign_toplevel_list
+//!     }
+//! }
+//!
+//! # let mut display = wayland_server::Display::<State>::new().unwrap();
+//! # let display_handle = display.handle();
+//! let mut state = State {
+//!     foreign_toplevel_list: ForeignToplevelListState::new::<State>(&display_handle),
+//! };
+//!
+//! let handle = state.foreign_toplevel_list.new_toplevel::<State>("Window Title", "com.example");
+//!
+//! // Handle can be used to update title and app_id
+//! handle.send_title("Window title has changed");
+//! handle.send_done();
+//!
+//! // Handle can also be used to close the window, after this call the handle will become inert,
+//! // and the handle will no longer be announced to clients
+//! handle.send_closed();
+//! ```
+
+use std::sync::{Arc, Mutex};
+
+use rand::distributions::{Alphanumeric, DistString};
+use wayland_protocols::ext::foreign_toplevel_list::v1::server::{
+    ext_foreign_toplevel_handle_v1::{self, ExtForeignToplevelHandleV1},
+    ext_foreign_toplevel_list_v1::{self, ExtForeignToplevelListV1},
+};
+use wayland_server::{
+    backend::{ClientId, GlobalId},
+    Client, DataInit, Dispatch, DisplayHandle, GlobalDispatch, New, Resource, Weak,
+};
+
+use crate::utils::user_data::UserDataMap;
+
+/// Handler for security context protocol
```

Comment:
Copy paste comment? ^^

Reply:
Actually more of a copy-paste of a whole module :laughing:
Fixed